src/redis_info_provider/info_poller.py

```diff
@@ -134,3 +134,6 @@ def _poll_shard(self, shard):
                     )
                 gevent.sleep(1)  # Cool-off period
                 continue  # Retry
+            except Exception as e:
```

Comment:
This contradicts [this comment](https://github.com/RedisLabs/redis_info_provider/blob/e60dfa0f4fc566c949d3b664bb27d83aacaa8afb/src/redis_info_provider/info_poller.py#L106) in the code.

I think swallowing exceptions forever might be the wrong approach, since we'll have no indication if something is seriously wrong (except spamming the logs). It might be more correct to have the info-provider _process_ crash if a greenlet raises an unexpected exception. I think this might even have been the intended behavior of `CCSShardWatcher.join`, just not implemented correctly.

Can you try to see how we can modify the code to support this behavior? ("Propagate an exception up from polling greenlets".)

Reply:
will try